src/btree/bt_discard.c

```diff
@@ -67,6 +67,9 @@ __wt_page_out(WT_SESSION_IMPL *session, WT_PAGE **pagep)
     page = *pagep;
     *pagep = NULL;
 
+    /* Assert we are not discarding a page that is being reconciled. */
+    WT_ASSERT(session, page->modify == NULL || !F_ISSET(page->modify, WT_PAGE_MODIFY_RECONCILING));
```

Comment:
The `modify == NULL` check here seems OK to me, but the __wt_page_is_modified function call is a more common pattern. Could you switch to that, or is there a reason this check should apply to clean (but previously modified) pages?

Reply:
Cleaned up these usages to leverage the existing functions.